File: medium/2486_append_characters_to_string_to_make_subsequence/solution.cpp
#include <string>

class Solution {
public:
  int appendCharacters(std::string s, std::string t) {
    const int sLen = s.size();
    const int tLen = t.size();

    int si = 0, ti = 0;

    while (si < sLen && ti < tLen) {
      if (s[si++] == t[ti])
        ++ti;
    }

    return tLen - ti;
  }
};

File: medium/2486_append_characters_to_string_to_make_subsequence/solution.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        s_len = len(s)
        t_len = len(t)

        si = ti = 0

        while si < s_len and ti < t_len:
            if s[si] == t[ti]:
                ti += 1

            si += 1

        return t_len - ti
